### PR TITLE
Fix broken links in a few data & story mdx files

### PR DESCRIPTION
**Pull request by dzole0311**
We have some broken links in a few `.mdx` files that point to an older `/datasets` route instead of the new `/data-catalog`. I additionally fixed the ones that pointed to `https://www.earthdata.nasa.gov/dashboard/eis/datasets` so that now they point to `https://www.earthdata.nasa.gov/dashboard/data-catalog` as I think at some point the `eis` routing part has been removed (@j08lue @sandrahoang686  please confirm)?

### CODE REVIEW

File: datasets/global-reanalysis-da.data.mdx [hunk 1]
```diff
@@ -368,7 +368,7 @@ The output variables available on VEDA include evapotranspiration (ET), gross pr
 <Block>
 <Prose>
 ## Explore the Data
-The global reanalysis is a large dataset with nearly two decades of daily output. Here we show a comparison of two dates for a single variable. We encourage users to <Link to='/datasets/global-reanalysis-da/explore'>Explore the Data></Link> to look at different dates and to compare variables.
+The global reanalysis is a large dataset with nearly two decades of daily output. Here we show a comparison of two dates for a single variable. We encourage users to <Link to='/data-catalog/global-reanalysis-da/explore'>Explore the Data></Link> to look at different dates and to compare variables.
```
Comment (sandrahoang686) [inline datasets/global-reanalysis-da.data.mdx:371]: There is a weird carrot after the word `Data`, dont think its supposed to be there. Maybe if we can move that too?

**Review by sandrahoang686 — APPROVED** [review]
Thank you for going through the code and cleaning 😌 🙌🏼

File: datasets/lis.da.trend.data.mdx [hunk 2]
```diff
@@ -90,7 +90,7 @@ layers:
 
      * [EIS Freshwater](https://freshwater.eis.smce.nasa.gov/)
      * [Land Information System](https://lis.gsfc.nasa.gov/)
-     * [Global Reanalysis Dataset page](https://www.earthdata.nasa.gov/dashboard/eis/datasets)
+     * [Global Reanalysis Dataset page](https://www.earthdata.nasa.gov/dashboard/data-catalog)
```
[the remark this answers is not in the document]
Comment (j08lue) [reply datasets/lis.da.trend.data.mdx:93]: Yes, this is related to 
- https://github.com/NASA-IMPACT/veda-config/issues/368

[the remark this answers is not in the document]
Comment (j08lue) [reply datasets/lis.da.trend.data.mdx:93]: And yes, it is correct to replace this with a taxonomy filter. 💯

File: stories/tws-trends.stories.mdx [hunk 3]
```diff
@@ -77,7 +77,7 @@ taxonomy:
 <Block>
   <Prose>
   ## Comparing the trends in water and carbon cycles
-  We applied a seasonal and trend decomposition algorithm to get the trend estimates for terrestrial water storage and gross primary production. The method can better help to deal with [nonstationarities](https://github.com/Earth-Information-System/sea-level-and-coastal-risk/blob/main/AMS_2023_Wanshu_Nie_for_VEDA_Discovery.pdf) and seasonal shifts and provide a more robust estimate of trends. These trend data sets are also provided in the [VEDA dataset page](https://www.earthdata.nasa.gov/dashboard/eis/datasets).
+  We applied a seasonal and trend decomposition algorithm to get the trend estimates for terrestrial water storage and gross primary production. The method can better help to deal with [nonstationarities](https://github.com/Earth-Information-System/sea-level-and-coastal-risk/blob/main/AMS_2023_Wanshu_Nie_for_VEDA_Discovery.pdf) and seasonal shifts and provide a more robust estimate of trends. These trend data sets are also provided in the [VEDA dataset page](https://www.earthdata.nasa.gov/dashboard/data-catalog).
```
[the remark this answers is not in the document]
Comment (j08lue) [reply stories/tws-trends.stories.mdx:81]: - https://github.com/NASA-IMPACT/veda-config/issues/368

**Pull request merged**